core/src/main/java/org/elasticsearch/action/admin/indices/validate/query/ShardValidateQueryRequest.java

```diff
@@ -95,6 +103,13 @@ public void readFrom(StreamInput in) throws IOException {
         explain = in.readBoolean();
         rewrite = in.readBoolean();
         nowInMillis = in.readVLong();
+        // TODO - reinstate this line once checkFieldNames is backported to 6.1 
+        // otherwise 7.0 commit will cause many BWC failures
+//        if (in.getVersion().onOrAfter(Version.V_6_1_0)) {
+        if (in.getVersion().after(Version.V_6_1_0)) {
```

Comment:
We will need to make sure this line reads `if (in.getVersion().onOrAfter(Version.V_6_1_0)) {` in the version that is backported to 6.1 too right?

Reply:
I'm actually wondering if it would be better to commit with the `onOrAfter` line and just accept the errors for a build or two. The last good commit stuff should mean that only the intake build fails on this. You could also set up the backport for 6.x branch before pushing the change on master so you can push both at the same time and minimise the chances of builds failing.